benchmark/solver/batch_solver.cpp

```diff
@@ -0,0 +1,517 @@
+    } else if (FLAGS_rhs_generation == "file") {
+        std::vector<gko::matrix_data<etype>> bdata(nbatch);
+        for (size_type i = 0; i < bdata.size(); ++i) {
+            std::string b_str;
+            if (FLAGS_batch_scaling == "implicit") {
+                b_str = "b_scaled.mtx";
+            } else {
+                b_str = "b.mtx";
+            }
+            std::string fname =
+                prob_string + "/" + std::to_string(i) + "/" + b_str;
+            std::ifstream b_fd(fname);
+            bdata[i] = gko::read_raw<etype>(b_fd);
+        }
+        auto temp_b = batch_vec<etype>::create(exec);
+        temp_b->read(bdata);
+        return batch_vec<etype>::create(exec, ndup, temp_b.get());
+    }
+    throw std::invalid_argument(std::string("\"rhs_generation\" = ") +
+                                FLAGS_rhs_generation + " is not supported!");
+}
+
+
+template <typename Engine>
+std::unique_ptr<batch_vec<etype>> generate_initial_guess(
+    std::shared_ptr<const gko::Executor> exec,
+    std::shared_ptr<const gko::BatchLinOp> system_matrix,
+    const batch_vec<etype> *rhs, Engine engine)
+{
+    auto nrhs = FLAGS_nrhs;
+    auto nbatch = FLAGS_num_batches;
+    auto ndup = FLAGS_num_duplications;
+    size_type multiplier = 1;
+    if (FLAGS_using_suite_sparse) {
+        multiplier = 1;
+    } else {
+        multiplier = ndup;
+    }
+    auto vec_size = gko::batch_dim<2>(
+        nbatch * multiplier,
+        gko::dim<2>{system_matrix->get_size().at(0)[1], nrhs});
+    if (FLAGS_initial_guess_generation == "0") {
+        return create_batch_matrix<etype>(exec, vec_size, gko::zero<etype>());
+    } else if (FLAGS_initial_guess_generation == "random") {
+        return create_batch_matrix<etype>(exec, vec_size, engine);
+    } else if (FLAGS_initial_guess_generation == "rhs") {
+        return rhs->clone();
+    }
+    throw std::invalid_argument(std::string("\"initial_guess_generation\" = ") +
+                                FLAGS_initial_guess_generation +
+                                " is not supported!");
+}
+
+
+void validate_option_object(const rapidjson::Value &value)
+{
+    if (!value.IsObject() || !value.HasMember("filename") ||
+        !value["filename"].IsString()) {
+        print_config_error_and_exit();
+    }
+}
+
+
+std::unique_ptr<gko::BatchLinOpFactory> generate_solver(
+    const std::shared_ptr<const gko::Executor> &exec,
+    const std::string &description)
+{
+    if (description == "richardson") {
+        using Solver = gko::solver::BatchRichardson<etype>;
+        return Solver::build()
+            .with_max_iterations(static_cast<int>(FLAGS_max_iters))
+            .with_rel_residual_tol(
+                static_cast<gko::remove_complex<etype>>(FLAGS_rel_res_goal))
+            .with_preconditioner(gko::preconditioner::batch::type::jacobi)
+            .with_relaxation_factor(
+                static_cast<gko::remove_complex<etype>>(0.95))
+            // FLAGS_relaxation_factor))
```

Comment:
I guess you need to re-enable this, because it might be good to briefly look at whether simple preconditioned Richardson can solve those systems and how it compares with GMRES etc.